utility_analysis_new/tests/contribution_bounders_test.py

```diff
@@ -96,6 +98,25 @@ def test_contribution_bounding_cross_partition_bounding_not_applied(self):
         # Check per- and cross-partition contribution limits are not enforced.
         self.assertEqual(set(expected_result), set(bound_result))
 
+    def test_contribution_bounding_cross_partition_bounding_and_sampling(self):
+        input = [("pid1", 'pk1', 1), ("pid1", 'pk1', 2), ("pid1", 'pk2', 3),
+                 ("pid1", 'pk2', 4), ("pid1", 'pk2', 5), ("pid1", 'pk3', 6),
+                 ("pid1", 'pk4', 7), ("pid2", 'pk4', 8)]
+        max_partitions_contributed = 3
+        max_contributions_per_partition = 5
+
+        bound_result = self._run_contribution_bounding(
+            input,
+            max_partitions_contributed,
+            max_contributions_per_partition,
+            partitions_sampling_prob=0.7)
+
+        # 'pk1' and 'pk2' are dropped by subsampling.
```

Comment:
This is very flaky, no? Could you please compute the flakiness of this and put it as comments?

Reply:
The test is deterministic, since ValueSampler is deterministic for the same partition key